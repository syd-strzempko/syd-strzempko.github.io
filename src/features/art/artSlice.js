import { createSlice } from '@reduxjs/toolkit';
import { importArt } from '../../utility/utils';

export const slice = createSlice({
    name: 'art',
    initialState: importArt(require.context('../../assets/art/', true, /\.jpg$/)),
    reducers: {},
});
  
// List selector

export const selectArts = (state) => {
    return state.art;
};

// Element selector

export const selectArt = (state, id) => {
    return state.art.find((single) => single.id.toString() === id);
};

export default slice.reducer;
  