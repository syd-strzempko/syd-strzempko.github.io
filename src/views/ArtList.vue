<template>
  <div class="content">
    <Header title="Art"></Header>
    <div class="wrapper">
      <ArtTile v-for="art in data.slice().reverse()" v-bind:key="art.id" v-bind:art="art" />
    </div>
  </div>
</template>

<script>
import Header from '@/components/Header.vue'
import ArtTile from '@/components/ArtTile.vue'

export default {
  props: {
    data: Array
  },
  name: 'ArtList',
  components: {
    Header,
    ArtTile
  }
};
</script>
<style scoped>
.wrapper {
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
}
</style>
