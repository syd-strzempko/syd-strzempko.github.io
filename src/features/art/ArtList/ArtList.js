import { Link } from 'react-router-dom';
import { useSelector } from 'react-redux';
import Layout from '../../../components/Layout';
import ProgressiveImage from '../../../components/ProgressiveImage';
import { selectArts } from '../artSlice';
import styles from '../art.module.css';

export const ArtList = () => {
  const arts = useSelector(selectArts);
  return (
    <Layout>
      <div className={styles.wrapper}>
        {arts?.map((art) => {
          return (
            <Link to={`/art/${art.id}`}>
              <div className={styles.tile} key={art.id}>
                <ProgressiveImage image={art} tile />
              </div>
            </Link>
          );
        })}
      </div>
    </Layout>
  );
}