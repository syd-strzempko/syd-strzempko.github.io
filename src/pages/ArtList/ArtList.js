import { Fragment } from 'react';
import Layout from '../../components/Layout';
import NavBar from '../../components/NavBar';
import './ArtList.css';

const ArtList = () => {
  return (
    <Layout>
      <NavBar />
      <Fragment>ArtIndex</Fragment>
    </Layout>
  );
}

export default ArtList;