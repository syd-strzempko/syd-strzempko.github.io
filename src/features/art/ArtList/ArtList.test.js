import { ArtList } from './ArtList';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import * as reactRedux from 'react-redux';
import { mockStore } from '../artSlice.test';

jest.mock('../../../components/Circles/Circles');

// Note: MemoryRouter allows for nested Link components to exist inside of & access stack of mocked router
describe('ArtList', () => {
    // Mock useSelectors
    beforeEach(() => { useSelectorMock.mockImplementation(selector => selector(mockStore)); });
    afterEach(() => { useSelectorMock.mockClear(); });

    const useSelectorMock = reactRedux.useSelector;

    // Unit tests
    test('renders correctly', () => {
        let view = renderer.create(
            <MemoryRouter>
                <ArtList />
            </MemoryRouter>
        ).toJSON();
        expect(view).toMatchSnapshot();
    });
});