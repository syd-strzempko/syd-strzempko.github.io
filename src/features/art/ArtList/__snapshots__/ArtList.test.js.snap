// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtList renders correctly 1`] = `
Array [
  <mock-Circles />,
  <div
    className="layout"
  >
    <div
      className="navWrapper"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        href="/work"
        onClick={[Function]}
      >
        Work
      </a>
      <a
        href="/art"
        onClick={[Function]}
      >
        Art
      </a>
    </div>
    <div
      className="navClose"
      onClick={[Function]}
    >
      X
    </div>
    <div
      className="navOpen visible"
      onClick={[Function]}
    >
      <img
        alt="menu"
        src="hamburger.png"
      />
    </div>
    <div
      className="wrapper"
    >
      <a
        href="/art/0"
        onClick={[Function]}
      >
        <div
          className="tile"
        >
          <img
            alt="work 0"
            className="tile"
            src="mock_long.jpg"
            style={Object {}}
          />
        </div>
      </a>
      <a
        href="/art/1"
        onClick={[Function]}
      >
        <div
          className="tile"
        >
          <img
            alt="work 1"
            className="tile"
            src="mock_long.jpg"
            style={Object {}}
          />
        </div>
      </a>
    </div>
  </div>,
]
`;
